Yoast/yoast-components
# 10267 add cornerstone toggle to sidebar

## Summary

This PR can be summarized in the following changelog entry:

* Adds a toggle to the Yoast Sidebar. The toggle is styled the same as the Gutenberg toggle. 

## Relevant technical choices:

* As WP components is not (yet) available, it was not possible to import the toggle component from WP and we had to copy the code into our own file named `GutenbergToggle.js`. When the WP component becomes available we can remove this code and simply import the component. 
* We did not include the SCSS file from WP in our code, as it is already loaded in WordPress when Gutenberg is enabled. This means the button works in the plugin sidebar, but does not display correctly in the standalone app. 
* The toggle button currently "toggles", for testing purposes. In the future this functionality needs to use the `yoast_wpseo_is_cornerstone` value. For this we need the the finished container which houses the cornerstone toggle and is connected to the store. 

## Test instructions

This PR can be tested by following these steps:

* Checkout out the branches `10267-add-cornerstone-toggle-to-sidebar` in both components as well as Wordpress SEO and yarn link them (not needed when this is an RC).
* Make sure `define( 'YOAST_FEATURE_GUTENBERG_SIDEBAR', true );` is set in wp-config.php. 
* Edit a post or page, look at the Yoast sidebar (under the plug icon) and see the toggle :). The toggle should switch on/off when clicked and it's styling should be the same as the Gutenberg toggles found under the settings/gear icon. 

Fixes #[10267](https://github.com/Yoast/wordpress-seo/issues/10267)
Requires #[10405](https://github.com/Yoast/wordpress-seo/pull/10405)
